### PR TITLE
RESOLVES #185 Adding dynamic calendar function

### LINKED ISSUE #185: Move create_dynamic_calendar_dialog from REVS scrubber over to FuncLib, add parameters
Here are the extra parameters I picture:

* No-go date range (an array I'd think)
* Disable weekends
* Custom text on dialog

Anything else?

### PR DESCRIPTION
**Pull request by CDPotter**
' PARAMETERS:
	' selected_dates_array  - is the output array it will contain dates in MM/DD/YY format
	' month_to_use          - this can be MM/YY or MM/DD/YY format as long as it is considered a date it will work. 
	' one_date_only         - this is a True/false parameter which will restrict the function to only allow one date to be selected if set to TRUE
	' disable_weekends      - this is a True/false parameter which will restrict the selection of weekends if set to TRUE
	' start_date & end_date - This will provide a range of dates which cannot be selected. These are to be entered as numbers. For example start_date = 3 and end_date = 14 the days 3 through 14 will be unavailable to select

### CODE REVIEW

File: MASTER FUNCTIONS LIBRARY.vbs [hunk 1]
```diff
@@ -1977,6 +1977,113 @@ Function create_panel_if_nonexistent()
...
+			OkButton 175, 170, 50, 15
+			CancelButton 225, 170, 50, 15
+			PushButton 85, 65, 20, 15, "<", prev_month_button					
+			PushButton 180, 65, 20, 15, ">", next_month_button
+		EndDialog
+	
+		IF one_date_only = TRUE THEN										'if only one date is allowed to be selected the script will act one way. Else it will allow for an large array of dates from a month to be build.
+			DO
+				selected_dates_array = ""									' declaring array at start of do loop. 
+				Dialog dialog1
```
Comment (theVKC) [inline MASTER FUNCTIONS LIBRARY.vbs]: The format we're using is simply `Dialog`, no `dialog1`.

File: MASTER FUNCTIONS LIBRARY.vbs [hunk 2]
```diff
@@ -1977,6 +1977,113 @@ Function create_panel_if_nonexistent()
...
+						END IF
+					END IF
+				NEXT
+				selected_dates_array = selected_dates_array & "end"						'this will allow us to delete the extra entry in the array
+				selected_dates_array = replace(selected_dates_array, ";end", "")
+				selected_dates_array = Split(selected_dates_array, ";")					'splitting array
+			IF Ubound(selected_dates_array) <> 0 AND (buttonpressed <> prev_month_button or buttonpressed <> next_month_button) THEN msgbox "Please select just one date."
+			LOOP until Ubound(selected_dates_array) = 0
+		ELSE
+			Dialog dialog1
```
Comment (theVKC) [inline MASTER FUNCTIONS LIBRARY.vbs]: Same here.

**Comment by theVKC** [comment]
See comments. Very minor issue.

Would this be incompatible entirely with the REVS scrubber interpretation? I would prefer to integrate and remove, it will be easier to manage that way long term, especially since we see this being helpful in other scripts.

**Comment by CDPotter** [comment]
I'd be hesitant, especially since REVS just went live and it's working smoothly in it's current form. Bit leave well enough alone mentality yes, but maybe down the line we can try to modernize it?

We'd have to dial back the switching months feature(adding another parameter), then make sure the output from the funclib version worked with what the custom version output since I changed a lot of the guts.

**Comment by theVKC** [comment]
`Dialog dialog` is wrong. Just `Dialog`. Plain, simple `Dialog`.

I really really want these to be merged together. I mean, I accept if you don't want to now, but I might try to get someone else to do it then.

**Comment by CDPotter** [comment]
I can take it back to the drawing board. I'm just expecting the influx from start of month tomorrow so I'll see how progress goes!

**Comment by theVKC** [comment]
Thanks for taking it @CDPotter! I appreciate your work on it!

**Pull request closed without merging**